src/static/js/product.js

```diff
@@ -56,11 +69,27 @@ $(function() {
                 idcolumn: "ID",
                 edit: function(row) {
                     tableform.fields_populate_from_json(dialog.fields, row);
+                    $("#productimage").prop("src", "image?db=asmtestdbdb&mode=media&id=" + row.MEDIAID);
+                    $("#productimage").closest("a").prop("href", "image?db=asmtestdbdb&mode=media&id=" + row.MEDIAID);
+                    $("#mediaid").val(row.MEDIAID);
+                    if (row.MEDIAID) {
+                        $("#removeimage").show();
+                    }
+                    else {
+                        $("#removeimage").hide();
+                    }
                     tableform.dialog_show_edit(dialog, row, {
-                        onchange: function() {
+                        onchange: async function() {
+                            let selectedfile = $("#imageinput")[0].files[0];
+                            if (selectedfile) {
+                                row.MEDIAID = await product.attach_image(selectedfile, "imageinput", row.ID);
+                            } else {
+                                row.MEDIAID = "";
```

Comment:
everywhere else uses a 0 for a foreign key that isn't linked to anything. Using empty string means inconsistent types. It still works fine for falsy tests